File: LeetCode/python/1065.py
class Solution(object):
    def indexPairs(self, text, words):
        """
        :type text: str
        :type words: List[str]
        :rtype: List[List[int]]
        """
        res = []
        for word in words:
            i = 0
            j = text.find(word, i)
            while j != -1:
                res.append([j, j + len(word) - 1])
                i = j + 1
                j = text.find(word, i)
        return sorted(res)

File: LeetCode/cpp/1065.cpp
class Solution {
public:
    vector<vector<int>> indexPairs(string &text, vector<string> &words) {
        vector<vector<int>> res;
        for (auto &word:words) {
            int i = 0;
            int pos = text.find(word);
            while (pos != -1) {
                res.push_back(vector<int>{pos, static_cast<int>(pos + word.size() - 1)});
                i = pos + 1;
                pos = text.find(word, i);
            }
        }
        sort(res.begin(), res.end());
        return res;
    }
};
